fotoapparat/src/main/java/io/fotoapparat/parameter/range/Range.java

```diff
@@ -0,0 +1,36 @@
+package io.fotoapparat.parameter.range;
+
+/**
+ * Created by ychen on 5/25/2017.
+ */
+
+public abstract class Range<T> {
+    abstract public boolean contains(T value);
+    abstract public T highest();
+    abstract public T lowest();
+
+    public String toString() {
+        return String.format("%1$s to %2$s", lowest(), highest());
+    }
+
+    public static <T> Range<T> emptyRange() {
+         return new EmptyRange<>();
+    }
+
+    public static class EmptyRange<T> extends Range<T> {
```

Comment:
Let's make it `private` for now. We can always make it `public` if we'll need to, but we won't be able to make it `private` again without breaking clients.